server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -1247,6 +1247,25 @@ private void executeSearch(
                 indicesAndAliases,
                 concreteLocalIndices
             );
+
+            // localShardIterators is empty since there are no matching indices. In such cases,
+            // we update the local cluster's status from RUNNING to SUCCESSFUL right away.
+            if (localShardIterators.isEmpty()
+                && clusters != SearchResponse.Clusters.EMPTY
+                && clusters.getCluster(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY) != null) {
```

Comment:
Question: when would the clusters list not include LOCAL_CLUSTER_GROUP_KEY and why we need to skip that case instead of adding successful entry?

Reply:
Here's my thought (@quux00 please correct me if my understanding is flawed): `localSharedIterators` can be empty if either a local cluster was not targeted in the first place or the specified index is nonexistent. If it's the former case, adding an entry would be incorrect. Moreover, any attempts to update its status would result in an NPE because the builder object "builds" upon the existing status object (where the status is set to `running` when the search op starts). I believe this is why a few tests were breaking when the check wasn't added initially.

Reply:
That's correct, Pawan. If the user never requested the local indices (it is a remote only CCS), then localShardIterators would be empty AND the local cluster alias ("") would be absent from the Clusters object. We need to handle the case where the user did request the local cluster, but no indices matched.

@pawankartik-elastic - can you update the code comment to explain that further, something like what I suggested earlier:
> If the local cluster was included in the search request by the client but had no matching indices, change its status from RUNNING to SUCCESSFUL and set total shards searched to zero.

Reply:
Thank you, Michael. And yes, I've further clarified the comment.